Responsive UI, Common Key validation and hash check, Download Activity Log.

Cereal-Killa opened the pull request.

FoxMcloud5655 commented:
Wow!  You're really putting some good work into this program.  Let me know when you are finished implimenting code, and I'll look it over.

I think it's definitely time for a new update.

Cereal-Killa commented on NUSGrabberGUI/Form1.cs:
```diff
@@ -193,49 +199,60 @@ private void STTitleList_SelectedIndexChanged(object sender, EventArgs e)
             {
                 if (v != null && v != "v0" && v != "" && v != " ")
                 {
-                    try { STVersionList.Items.Add(v.TrimStart(' ').Substring(0, v.LastIndexOf('(') - 1)); }
```
The exception thrown here was causing the system titles to take a long time to filter.

Cereal-Killa commented on NUSGrabberGUI/Form1.cs:
```diff
@@ -191,49 +199,60 @@ private void STTitleList_SelectedIndexChanged(object sender, EventArgs e)
         }
 
         private void STSearchBox_TextChanged(object sender, EventArgs e)
         {
```
It looks like there were a lot of changes here however it's just the try catch that made the git diff to look huge, all I did was to add a try/catch block to block the UI and the beginupdate()/endupdate() calls.

Cereal-Killa commented:
I screwed up reverting a commit which contained other changes, trying to recall the reverted commit now.

Cereal-Killa commented:
I think I got it back, it looks cleaner now.

Cereal-Killa commented:
Added a few more things...

1. Optimized the initial list parsing (was incredibly slow in development due to the many thrown exceptions).

2. If the folder is not empty and can't be cleared we'll open it using Explorer to make the user's life easier.

3. Added a download log window, basic initial version.

Cereal-Killa commented:
![2018-07-27_17-44-58](https://user-images.githubusercontent.com/25939765/43346190-9b07dc16-91c6-11e8-9012-58b54425be24.png)

Cereal-Killa commented:
Btw, I'm done for this pull, thanks for taking some time to review.

FoxMcloud5655 commented:
There's a lot of work to be done to this code to make it ready for deployment (code structure, optimizing, language versatility, debug log writing, etc.), but these are great!

I will start working on merging this and have it ready for deployment soon.

FoxMcloud5655 commented:
I ended up not really liking all the changes...  Sorry about that.  I'm still happy you thought it was good enough to edit on your own, though :)
